### ISSUE TITLE
Potential validation/testing leakage due to the absence of switching between train and eval modes

### ISSUE THREAD

**Issue by jinhaoduan** [icon-lab/SynDiff#46]
Thanks for releasing the code. 

I noticed that there's no switching between `model.train()` mode and `model.eval()` mode when transitioning between the training and validation processes. Could this potentially lead to leakage of validation/test information (BN will track the statistics in the validation/testing set)?

Another minor issue is the incorrect parameter in https://github.com/icon-lab/SynDiff/blob/8e5064e6f1a798a166cee5e98c66700bab92b0cc/train.py#L699 The generator should be `gen_diffusive_2` for T1 -> T2, right?

**Comment by tcukur** [comment]
We use three-way data split, training and validation are to fit model and tune hyperparameters. Since we use batch size of 1, we actually induce instance normalization, should be relatively resilient against leakage. For test, each image is evaluated on its own, so shouldn't cause leakage.
